### PR TITLE
[vfs] [posix] PosixFile::Copy(): improve coping between filesystems

### PR DESCRIPTION
**Pull request by Karlson2k**
This is an improvement for PR #6582.
More error logging, direct coping instead of running through CFile wrapper.

Little bit more chances of success by opening target with `O_WRONLY`.

### CODE REVIEW

**Comment by Karlson2k** [comment]
jenkins build this please

File: xbmc/filesystem/posix/PosixFile.cpp [hunk 1]
```diff
@@ -317,6 +318,55 @@ bool CPosixFile::Rename(const CURL& url, const CURL& urlnew)
...
+      portion_size = read(src, buf.get(), buf.size());
+    }
+    close(dst);
+    close(src);
+
+    if (portion_size == 0)
+    {
+      if (unlink(name.c_str()) < 0) // remove original file
+        CLog::LogF(LOGWARNING, "Error removing source file \"%s\" after coping", name.c_str());
+      return true;
```
Comment (wsnipex) [inline xbmc/filesystem/posix/PosixFile.cpp]: this is not correct. if unlink fails, rename/move failed.

Comment (Karlson2k) [inline xbmc/filesystem/posix/PosixFile.cpp]: See discussion here: https://github.com/xbmc/xbmc/pull/6582#discussion_r25683810

**Comment by Karlson2k** [comment]
Updated with "atomic move" as requested by @wsnipex, @topfs2, @notspiff and added additional check.

jenkins build this please

**Comment by Karlson2k** [comment]
Will merge it in two days if no objections.

**Comment by topfs2** [comment]
Personally I prefer the other PR. It feels cleaner and could be moved up to either a super class or in our cfile, and then work cross vfs implementations.

**Comment by Karlson2k** [comment]
Updated to be improvement.
List of improvements:
* More error resistant
* More error checks
* More log reports - simpler resolve endusers' problems
* Faster start (instantly - all CURLs are already decoded to local pathnames), no need to create additional two CFile object, decode two URLs, creating two CPosixFile objects, decode two URLs to local pathnames
* Faster copy: 1MB buffer better suits local files than 128KB, no need to update file pointers in CPosixFile for each read and for each write, no need to update posix_fadvise position for each read operation, no need to reset global screensaver before each read, no need to update local timer
* Leave a copy of file if original file was damaged during delete
* Little bit more chances of success by opening target with `O_WRONLY`

**Comment by notspiff** [comment]
just merge this shit already. i retract mine. may the NIH be with you.

**Comment by Karlson2k** [comment]
@notspiff I merged your version already. :)
Thanks for positive response! ;)

**Comment by Memphiz** [comment]
@notspiff the National Institute of Health?

**Comment by fritsch** [comment]
http://en.wikipedia.org/wiki/Not_invented_here <- NIH

**Comment by MartijnKaijser** [comment]
so what's status on this one? iirc the other fix was merged right and this can be closed?

**Comment by Karlson2k** [comment]
Now this is improvment. List of improvmnet is here: https://github.com/xbmc/xbmc/pull/6584#issuecomment-81496944

**Comment by MartijnKaijser** [comment]
If still needed please open a new PR and reference this one.

**Pull request closed without merging**